sqlalchemy-stubs/sql/elements.pyi

```diff
@@ -44,27 +45,32 @@ class ClauseElement(Visitable):
     def __bool__(self): ...
     __nonzero__: Any = ...
 
+_CE = TypeVar('_CE', bound=ColumnElement)
+
 class ColumnElement(operators.ColumnOperators, ClauseElement, Generic[_T]):
     __visit_name__: str = ...
-    primary_key: Any = ...
-    foreign_keys: List[Any] = ...
-    key: Any = ...
-    def self_group(self, against: Optional[Any] = ...): ...
+    primary_key: bool = ...
+    foreign_keys: List[ForeignKey] = ...
+    key: Optional[str] = ...
+    def self_group(self: _CE, against: Optional[Any] = ...) -> Union[AsBoolean, Grouping[_T], _CE]: ...
```

Comment:
Normally we try to avoid such unions in return types. But this one is probably OK.